pkg/controller/v1alpha2/experiment/util/status_util.go

```diff
@@ -0,0 +1,135 @@
+
+	isObjectiveGoalReached := updateTrialsSummary(instance, trials)
+
+	updateExperimentStatusCondition(instance, isObjectiveGoalReached)
+
+}
+
+func updateTrialsSummary(instance *experimentsv1alpha2.Experiment, trials *trialsv1alpha2.TrialList) bool {
+
+	var trialsPending, trialsRunning, trialsSucceeded, trialsFailed, trialsKilled int
+	var bestTrialIndex int
+	var bestTrialValue float64
+	isObjectiveGoalReached := false
+	objectiveMetricName := instance.Spec.Objective.ObjectiveMetricName
+	objectiveValueGoal := instance.Spec.Objective.Goal
+	objectiveType := instance.Spec.Objective.Type
+	for index, trial := range trials.Items {
+		if trial.IsKilled() {
+			trialsKilled++
+		} else if trial.IsFailed() {
+			trialsFailed++
+		} else if trial.IsSucceeded() {
+			trialsSucceeded++
+		} else if trial.IsRunning() {
+			trialsRunning++
+		} else {
+			trialsPending++
+		}
+
+		for _, metric := range trial.Status.Observation.Metrics {
+			if objectiveMetricName == metric.Name {
+				if objectiveType == experimentsv1alpha2.ObjectiveTypeMinimize {
+					if bestTrialValue < metric.Value {
+						bestTrialValue = metric.Value
+						bestTrialIndex = index
+					}
+					if bestTrialValue <= objectiveValueGoal {
+						isObjectiveGoalReached = true
+					}
+				} else if objectiveType == experimentsv1alpha2.ObjectiveTypeMaximize {
+					if bestTrialValue > metric.Value {
+						bestTrialValue = metric.Value
+						bestTrialIndex = index
+					}
+					if bestTrialValue >= objectiveValueGoal {
+						isObjectiveGoalReached = true
+					}
+				}
+			}
+		}
+	}
+	if len(trials.Items) > 0 {
+		instance.Status.TrialsPending = trialsPending
+		instance.Status.TrialsRunning = trialsRunning
+		instance.Status.TrialsSucceeded = trialsSucceeded
+		instance.Status.TrialsFailed = trialsFailed
+		instance.Status.TrialsKilled = trialsKilled
+
+		bestTrial := trials.Items[bestTrialIndex]
+
+		instance.Status.CurrentOptimalTrial.ParameterAssignments = []trialsv1alpha2.ParameterAssignment{}
+		for _, parameterAssigment := range bestTrial.Spec.ParameterAssignments {
+			instance.Status.CurrentOptimalTrial.ParameterAssignments = append(instance.Status.CurrentOptimalTrial.ParameterAssignments, parameterAssigment)
+		}
+
+		instance.Status.CurrentOptimalTrial.Observation.Metrics = []trialsv1alpha2.Metric{}
+		for _, metric := range bestTrial.Status.Observation.Metrics {
+			instance.Status.CurrentOptimalTrial.Observation.Metrics = append(instance.Status.CurrentOptimalTrial.Observation.Metrics, metric)
+		}
+	}
+	return isObjectiveGoalReached
+}
+
+func updateExperimentStatusCondition(instance *experimentsv1alpha2.Experiment, isObjectiveGoalReached bool) {
+
+	completedTrialsCount := instance.Status.TrialsSucceeded + instance.Status.TrialsFailed + instance.Status.TrialsKilled
+	failedTrialsCount := instance.Status.TrialsFailed
+
+	if (instance.Spec.MaxTrialCount != nil) && (completedTrialsCount >= *instance.Spec.MaxTrialCount) {
```

Comment:
I think we can mark a experiment as ExperimentStatusSucceeded in two cases:
1. ObjectiveGoalReached;
2. instance.Status.TrialsSucceeded >= *instance.Spec.MaxTrialCount

Reply:
I have added the change with different message for both these use cases